plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchangeSource.java

```diff
@@ -11,46 +11,46 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.trino.server.testing.exchange;
+package io.trino.plugin.exchange;
 
 import com.google.common.collect.ImmutableList;
-import io.airlift.slice.InputStreamSliceInput;
 import io.airlift.slice.Slice;
 import io.airlift.slice.SliceInput;
-import io.airlift.units.DataSize;
 import io.trino.spi.exchange.ExchangeSource;
 
 import javax.annotation.Nullable;
 import javax.annotation.concurrent.GuardedBy;
+import javax.crypto.SecretKey;
 
-import java.io.FileInputStream;
-import java.io.FileNotFoundException;
+import java.io.IOException;
 import java.io.UncheckedIOException;
-import java.nio.file.Path;
+import java.net.URI;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Optional;
 import java.util.concurrent.CompletableFuture;
 
-import static io.airlift.units.DataSize.Unit.KILOBYTE;
-import static java.lang.Math.toIntExact;
 import static java.util.Objects.requireNonNull;
 
-public class LocalFileSystemExchangeSource
+public class FileSystemExchangeSource
         implements ExchangeSource
 {
-    private static final int BUFFER_SIZE_IN_BYTES = toIntExact(DataSize.of(4, KILOBYTE).toBytes());
-
+    private final FileSystemExchangeStorage exchangeStorage;
+    @GuardedBy("this")
+    private final Iterator<URI> files;
     @GuardedBy("this")
-    private final Iterator<Path> files;
+    private final Iterator<Optional<SecretKey>> secretKeys;
 
     @GuardedBy("this")
     private SliceInput sliceInput;
     @GuardedBy("this")
     private boolean closed;
 
-    public LocalFileSystemExchangeSource(List<Path> files)
+    public FileSystemExchangeSource(FileSystemExchangeStorage exchangeStorage, List<URI> files, List<Optional<SecretKey>> secretKeys)
     {
+        this.exchangeStorage = requireNonNull(exchangeStorage, "exchangeStorage is null");
         this.files = ImmutableList.copyOf(requireNonNull(files, "files is null")).iterator();
+        this.secretKeys = ImmutableList.copyOf(requireNonNull(secretKeys, "secretKeys is null")).stream().iterator();
```

Comment:
assertThat `files` and `secretKeys` lists has same lenght.
Also consider creating a wrapper class which contains URI and Optional<SecretKey> and pass just as single list.

Reply:
Added assertion.